Light/src/rendering/renderer.cpp

```diff
@@ -60,6 +70,41 @@ namespace Light
 			}
 		}
 
+		for (int i = 0; i < 4; i++)
+		{
+			if (i < (int)s_sceneData->spotLights.size())
+			{
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].position", glm::vec4(s_sceneData->spotLights[i].position, 1.0));
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].color", glm::vec4(s_sceneData->spotLights[i].color, 1.0));
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].direction", glm::vec4(s_sceneData->spotLights[i].direction, 1.0));
+				shader->setUniformFloat("u_spotLights[" + std::to_string(i) + "].innerCutoff", s_sceneData->spotLights[i].innerCutoff);
+				shader->setUniformFloat("u_spotLights[" + std::to_string(i) + "].outerCutoff", s_sceneData->spotLights[i].outerCutoff);
+			}
+			else
+			{
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].position", glm::vec4(0.0, 0.0, 0.0, 1.0));
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].color", glm::vec4(0.0, 0.0, 0.0, 1.0));
+				shader->setUniformVec4("u_spotLights[" + std::to_string(i) + "].direction", glm::vec4(0.0, 0.0, 0.0, 1.0));
+				shader->setUniformFloat("u_spotLights[" + std::to_string(i) + "].innerCutoff", 0.0);
+				shader->setUniformFloat("u_spotLights[" + std::to_string(i) + "].outerCutoff", 0.0);
+			}
+		}
+
+		for (int i = 0; i < 4; i++)
+		{
+			if (i < (int)s_sceneData->directionalLights.size())
+			{
+				shader->setUniformVec4("u_directionalLights[" + std::to_string(i) + "].direction", glm::vec4(s_sceneData->directionalLights[i].direction, 0.0));
+				shader->setUniformVec4("u_directionalLights[" + std::to_string(i) + "].color", glm::vec4(s_sceneData->directionalLights[i].color, 1.0));
+			}
+			else
+			{
+				shader->setUniformVec4("u_directionalLights[" + std::to_string(i) + "].direction", glm::vec4(0.0, 0.0, 0.0, 0.0));
+				shader->setUniformVec4("u_directionalLights[" + std::to_string(i) + "].color", glm::vec4(0.0, 0.0, 0.0, 1.0));
+			}
+		}
+
+		shader->setUniformVec3("cameraPosition", s_sceneData->cameraPosition);
 		shader->setUniformInt("u_numPointLights", (int)s_sceneData->pointLights.size());
```

Comment:
Same comment as before about the typecast.

Reply:
The typecast was already present, like I mentioned earlier